x/pairing/keeper/scores/score.go

```diff
@@ -0,0 +1,230 @@
+//		Data structure:
+//			The PairingSlot object holds a map of requirements. The map keys are the requirement object's type (
+//			an object that satisfies the ScoreReq interface) and the map values are the requirement object
+//			itself. The slots are assigned with requirements in the CalcSlots() function.
+//
+// 	- Pairing score:
+//		What is it?
+// 			The Pairing score defines the score for each provider when picking providers to pair. The score is
+//			calculated for all providers for each slot. It depends on the slot requirements since the score of
+//			a provider can change with different requirements. For example, say that slot A requires a provider from
+//			Europe and slot B requires a provider from Asia. We have 2 possible providers: provider A which operates
+//			from Europe and and provider B that operates from Asia. Clearly, provider A's score for the Europe slot
+//			will be better than provider B's score for the Europe slot.
+//		Data structure:
+//			The PairingScore object is created for each provider, holding a pointer to the provider, its final pairing
+//			score and the score components (the provider's stake score, geo score, etc.). The ScoreComponents map keys
+//			are the requirement object's type and the map values are the score for this requirement.
+//
+// 	- Pairing score stratgy:
+//		What is it?
+// 			A Pairing score strategy (or just strategy) defines the weight of each score component in the
+// 			final score calculation for each provider. The calculation is done as follows: req1^w1 + req2^w2 + ...
+//			For example, say that the score components are stake and geolocation. To make the pairing process
+// 			biased towards stake, we could define the strategy: w_stake = 2, w_geo = 1. So the final score
+//			will be providerStakeScore^2 + providerGeoScore^1.
+//		Data structure:
+//			The ScoreStrategy object is a map in which the keys are the requirement object's type and the values
+//			are the requirement's weight.
+//
+//	Adding new requirements:
+// 	  To add a new requirement, one should create a new object that satisfies the ScoreReq interface and update the
+//	  CalcSlots() function so the new requirement will be assigned to the pairing slots (according to some logic).
+//	  Lastly, append the new requirement object to the allReqTypes var in the init() function in this file. Use
+//	  StakeReq or GeoReq as examples.
+
+package scores
+
+import (
+	"fmt"
+	"math/big"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/lavanet/lava/utils"
+	epochstoragetypes "github.com/lavanet/lava/x/epochstorage/types"
+	planstypes "github.com/lavanet/lava/x/plans/types"
+	tendermintcrypto "github.com/tendermint/tendermint/crypto"
+)
+
+var (
+	uniformStrategy ScoreStrategy
+	allReqNames     []string
+)
+
+// TODO: currently we'll use weight=1 for all reqs. In the future, we'll get it from policy
+func init() {
```

Comment:
you can initialize these directly and drop the `init()` function. e.g.
```
var allReqNames = []string{
    stake_req_ name,
}

var uniformStrategy = ScoreStrategy{ 
    stake_req_name: 1,
}
```

Reply:
kept the init function so I can verify that len(allReqs) == len(strategy). see https://github.com/lavanet/lava/commit/c2ce6a219214545c35ff7e455b49f6f3a89152e0

Reply:
you can define them staticly and only check length, it's going to be ocmpiled this way so it's more memory efficient and less code to read

Reply:
fixed